data_processing/process_data.py

```diff
@@ -0,0 +1,27 @@
+# This script handles the processing and creation of the interaction matrix
+
+import numpy as np
+
+def create_interaction_matrix(directory_path, num_users, num_movies, files):
+    interaction_matrix = np.zeros((num_users, num_movies), dtype=np.int8)
+    user_mapper = {}
+    current_user_index = 0
+    
+    for filename in files:
+        with open(os.path.join(directory_path, filename), 'r') as file:
+            movie_id = int(file.readline().strip().replace(':', ''))
+            movie_idx = movie_id - 1
+            
+            for line in file:
+                user_id, rating, _ = line.strip().split(',')
+                user_id = int(user_id)
+                rating = int(rating)
+                
+                if rating in (4, 5):
```

Comment:
You can use a pattern to "reverse" your if statements and return early. Instead of:

```
  if condition_a:
     code = blah + foo
     if condition_b:
        some_code = foo + bar
        other_thing()
```

You can write:
```
  if not condition_a:
    return/exit/continue
  code = blah + foo

  if not condition_b:
    return/exit/continue

  some_code = foo + bar
  other_thing()
```

This means you have less levels of indentation and makes your code easier to read. The person reading it doesn't have to think "Okay so X is true and Y is true if we're in this block".

In this case, you would reverse the if condition and `continue` the for loop.

Reply:
Okay